providers/oidc.go

```diff
@@ -68,6 +71,13 @@ func (p *OIDCProvider) Redeem(ctx context.Context, redirectURL, code string) (*s
 // EnrichSession is called after Redeem to allow providers to enrich session fields
 // such as User, Email, Groups with provider specific API calls.
 func (p *OIDCProvider) EnrichSession(ctx context.Context, s *sessions.SessionState) error {
+	if p.IntrospectURL.String() != "" {
```

Comment:
I feel like this should be last in the ordering of enrichments. The no-email failure scenario is a hard error & return since our framework requires an email.